lang/python/python3-package.mk

```diff
@@ -63,6 +63,12 @@ define Py3Package
     endef
   endif
 
+  ifndef Py3Package/$(1)/extra_install
```

Comment:
this seems a bit redundant, unless there is some ordering requirement, where `extra_install` needs to run after `$(SHELL) $(python3_mk_path)python-package-install.sh "3" \`

otherwise the `$(call Py3Package/$(1)/install,$$(1))` could handle most cases ;
we could move that call after the ``$(SHELL) $(python3_mk_path)python-package-install.sh "3" \`` call, if that helps
using `Py3Package/$(1)/filespec` & `Py3Package/$(1)/install` are no mutually exclusive

i was also thinking maybe of adding some `Py3Package/$(1)/preinst` & `Py3Package/$(1)/postinst` hooks, but I guess the main ones from core [ Package/$(1)/{preinst,postinst} ]  should be able to handle things well as they are

Reply:
I wanted to use the P3Package install but add on additional (normal package-type) install stuff ... in particular this helps for Python apps like Radicale 2.x which needs to app stuff that isn't part of Py3Package, but AFAICT there is no way to do that without copy&pasting form P3Package into a custom install.

Reply:
i'm a bit vague/confused about `hon apps like Radicale 2.x which needs to add stuff that isn't part of Py3Package` ;

one example of using `Py3Package/$(1)/install` is https://github.com/openwrt/packages/blob/master/lang/python/python3/Makefile#L254
`python3-base` uses this to install the Python binaries and it could also install other things that typically go into `Package/$(1)/install`

could you exemplify what you have in mind with this ?

Reply:
So I have 

```
define Py3Package/radicale2/extra_install
        $(SED) 's,#!.*,#!/usr/bin/python$(PYTHON3_VERSION),g' $(1)/usr/bin/radicale
        $(MV) $(1)/usr/bin/radicale $(1)/usr/bin/radicale2
        $(INSTALL_DIR) $(1)/etc/config
        $(INSTALL_CONF) ./files/radicale2.config $(1)/etc/config/radicale2
        $(INSTALL_BIN) ./files/radicale2.init $(1)/etc/init.d/radicale2
endef
```
which is effectively
```
        $(call Py3Package/$(1)/install,$$(1))
        find $(PKG_INSTALL_DIR) -name "*\.exe" | xargs rm -f
        $(SHELL) $(python3_mk_path)python-package-install.sh "3" \
                "$(PKG_INSTALL_DIR)" "$$(1)" \
                "$(HOST_PYTHON3_BIN)" "$$(2)" \
                "$$$$$$$$$$(call shvar,Py3Package/$(1)/filespec)"
        $(SED) 's,#!.*,#!/usr/bin/python$(PYTHON3_VERSION),g' $(1)/usr/bin/radicale
        $(MV) $(1)/usr/bin/radicale $(1)/usr/bin/radicale2
        $(INSTALL_DIR) $(1)/etc/config
        $(INSTALL_CONF) ./files/radicale2.config $(1)/etc/config/radicale2
        $(INSTALL_BIN) ./files/radicale2.init $(1)/etc/init.d/radicale2
```

**Ahhh!** Now I understand ... yeah reordering Py3Package/$(1)/install would do the trick.

I'll change that.

Reply:
Done